File: [23.12]Design-Pattern/Java/src/creational/prototpye/Drink.java
package creational.prototpye;

import java.util.ArrayList;
import java.util.List;

public class Drink implements Cloneable {
    private List<Integer> list = new ArrayList<>();

    public Drink(List<Integer> list) {
        this.list = list;
    }

    @Override
    protected Object clone() throws CloneNotSupportedException {
        List<Integer> copyList = new ArrayList<>(list);
        return new Drink(copyList);
    }

    @Override
    public String toString() {

        return "Drink{" +
                "list = " + System.identityHashCode(list) + list +
                '}';
    }
}

File: [23.12]Design-Pattern/Kotlin/src/main/kotlin/prototype/Drink.kt
package prototype

data class Drink(val list: List<Int>) {

    fun copy(): Drink{
        val copyList = list.toMutableList()
        return Drink(copyList)
    }
    
    override fun toString(): String {
        val hashCode = System.identityHashCode(list)
        return "Drink(list=$list, $hashCode)"
    }
}